tests/check_framework/test_security.py

```diff
@@ -504,3 +504,26 @@ def test_failure_view_invalid_signature(self):
             csrf.check_csrf_failure_view(None),
             [Error(msg, id='security.E101')],
         )
+
+
+class CheckTypeOfAllowedHostsTest(SimpleTestCase):
```

Comment:
We can make these a little more 'DRY' by splitting into just two tests (one for valid, one for invalid) and then use sub-tests. Something along the lines of:
```
    def test_allowed_hosts_invalid(self):
        tests = (
            'example.com',
            ['example', 1],
            None
        )
        for value in tests:
            with self.subTest(value=value), override_settings(ALLOWED_HOSTS=value):
                self.assertEqual(
                    base.check_type_of_allowed_hosts(None),
                    [base.W024]
                )
```
You can then easily add in `[]` to the list of valid (as far as this test is concerned) values.

Reply:
Sounds reasonable; I've yet to use sub test so this will be good 👍

Reply:
Updated in https://github.com/django/django/pull/14149/commits/aa6c4f49569a709e457d8177ca35d544cb6da7bb